testsuite/MDAnalysisTests/coordinates/test_writer_api.py

```diff
@@ -0,0 +1,102 @@
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import absolute_import
+
+import itertools
+import pytest
+
+import MDAnalysis as mda
+
+
+# grab all known writers
+# sort so test order is predictable for parallel tests
+writers = sorted(set(mda._MULTIFRAME_WRITERS.values()) |
+                 set(mda._SINGLEFRAME_WRITERS.values()),
+                 key=lambda x: x.__name__)
+known_ts_haters = [
+    mda.coordinates.MOL2.MOL2Writer,
+    mda.coordinates.PDB.PDBWriter,
+    mda.coordinates.PDB.MultiPDBWriter,
+    mda.coordinates.PQR.PQRWriter,
+    mda.coordinates.PDBQT.PDBQTWriter,
+    mda.coordinates.LAMMPS.DATAWriter,
+    mda.coordinates.CRD.CRDWriter,
+]
+
+
+@pytest.mark.parametrize('writer', [w for w in writers
+                                    if not w in known_ts_haters])
+def test_ts_deprecated(writer, tmpdir):
+    u = mda.Universe.empty(10, trajectory=True)
+
+    if writer == mda.coordinates.chemfiles.ChemfilesWriter:
+        # chemfiles Writer exists but doesn't work without chemfiles
+        if not mda.coordinates.chemfiles.check_chemfiles_version():
+            pytest.skip("Chemfiles not available")
+        fn = str(tmpdir.join('out.xtc'))
+    else:
+        fn = str(tmpdir.join('out.traj'))
+
+    with writer(fn, n_atoms=u.atoms.n_atoms) as w:
+        with pytest.warns(DeprecationWarning):
+            w.write(u.trajectory.ts)
+
+
+@pytest.mark.parametrize('writer', writers)
+def test_write_with_atomgroup(writer, tmpdir):
+    u = mda.Universe.empty(10, trajectory=True)
+
+    if writer == mda.coordinates.chemfiles.ChemfilesWriter:
+        # chemfiles Writer exists but doesn't work without chemfiles
+        if not mda.coordinates.chemfiles.check_chemfiles_version():
+            pytest.skip("Chemfiles not available")
+        fn = str(tmpdir.join('out.xtc'))
+    elif writer == mda.coordinates.MOL2.MOL2Writer:
+        pytest.skip("MOL2 only writes MOL2 back out")
+    elif writer == mda.coordinates.LAMMPS.DATAWriter:
+        pytest.skip("DATAWriter requires integer atom types")
+    else:
+        fn = str(tmpdir.join('out.traj'))
+
+    with writer(fn, n_atoms=u.atoms.n_atoms) as w:
+        w.write(u.atoms)
+
+
+@pytest.mark.parametrize('writer', writers)
+def test_write_with_universe(writer, tmpdir):
+    u = mda.Universe.empty(10, trajectory=True)
+
+    if writer == mda.coordinates.chemfiles.ChemfilesWriter:
+        # chemfiles Writer exists but doesn't work without chemfiles
+        if not mda.coordinates.chemfiles.check_chemfiles_version():
+            pytest.skip("Chemfiles not available")
+        fn = str(tmpdir.join('out.xtc'))
+    elif writer == mda.coordinates.MOL2.MOL2Writer:
+        pytest.skip("MOL2 only writes MOL2 back out")
```

Comment:
Can we have add two separate tests here that write atomgroups and universes to MOL2 (from MOL2 input) and DATA (from LAMMPS input)?

Reply:
Quick look here seems to indicate that MOL2 will work with an atomgroup but will fail with a universe (testing with mol2_comments_header).